### ISSUE TITLE
Expose action manager to Excalidraw API

### ISSUE THREAD

**Issue by ad1992** [excalidraw/excalidraw#3007]
We should expose the action manager via excalidraw ref API so the host can use the actions directly whenever needed,
 
Actions like `zoomToFit`, `zoomToFitAllElements`, `goToCollaborator` and many more would be useful for host to use.

With this host will have a good control over the actions. 
 
 cc @dwelle

**Comment by dwelle** [comment]
Yep, though we'll have to export all the actions. Right now you execute action using its exported object. While we could support supplying names, I was actually planning to remove the names coz they only introduce complexity and intellisense doesn't work with them.

**Comment by ad1992** [comment]
@dwelle you mean maintaining a map of name and action object so the host can use the name to execute the action like `excalidrawRef.actionManager.execute(actionName)` ?
As for the intellisence we can have an object `ACTION` like 👇 
```js
const ACTION = {
    ZOOM_TO_FIT: 'zoomToFit',
    GO_TO_COLLABORATOR: 'goToCollaborator'
}
```

**Comment by dwelle** [comment]
The opposite. We currently use [string names for actions](https://github.com/excalidraw/excalidraw/blob/ecbd5ba55d6af86dcc533dda2242ccb134bac570/src/actions/types.ts#L25), while to execute them, we supply the action object. I want to get rid  of the string names altogether.

**Comment by ad1992** [comment]
We can use action object for executing and [rendering](https://github.com/excalidraw/excalidraw/blob/ecbd5ba55d6af86dcc533dda2242ccb134bac570/src/actions/manager.tsx#L103) but we still might want to keep the name field to act as a differentiator, example I am using it [here](https://github.com/excalidraw/excalidraw/blob/e6cd97c4f285669d6b4ec28f693bd33bed3190b1/src/actions/manager.tsx#L72). 

However, I still think we should support executing action via names instead of exported object to keep the API simple and keep the names as constants so it's easy to use.

**Comment by ad1992** [comment]
I meant lets only allow by name so the host doesn't have to import the action as well to execute and API will also be simple.

And we will still need the name to act as a differentiator from other actions right as mentioned above.

**Comment by dwelle** [comment]
> I meant lets only allow by name so the host doesn't have to import the action as well to execute and API will also be simple.

Why? Exporting the actions will not add that many bytes, and it'll be much more friendlier both to us and the host apps (intellisense-wise).

> And we will still need the name to act as a differentiator from other actions right as mentioned above.

I don't understand.

**Comment by ad1992** [comment]
> > I meant lets only allow by name so the host doesn't have to import the action as well to execute and API will also be simple.
> 
> Why? Exporting the actions will not add that many bytes, and it'll be much more friendlier both to us and the host apps (intellisense-wise).

I am not sure why it would be better (or maybe I am missing something) as we can export a constant which includes the name for all actions, then we will get the IntelliSense. 
> 
> > And we will still need the name to act as a differentiator from other actions right as mentioned above.
> 
> I don't understand.

I was talking about this [usecase](https://github.com/excalidraw/excalidraw/blob/e6cd97c4f285669d6b4ec28f693bd33bed3190b1/src/actions/manager.tsx#L72). 

API wise I feel passing name will be simpler but If we want to go with exporting actions, I still feel removing name completely might not be possible as for the above case. We would need some differentiator to distinguish when there is dependency among actions.

**Comment by dwelle** [comment]
The use case you link to can be achieved by comparing to the action objects, which will be better anyway coz it'll be explicit, typed, and more accessible.

And that exact code path is actually what's causing the "actions are being prevented in view mode", so let's wait until we fix it before we continue discussing whether string action names have any benefit.

**Comment by kiraaziz** [comment]
Is there a solution for this problem yet ?

**Comment by hakotaco** [comment]
Is there any movement on this whatsoever? This would be a really useful addition.
